formula-android/src/main/java/com/instacart/formula/android/internal/ActivityStoreContextImpl.kt

```diff
@@ -72,8 +72,12 @@ internal class ActivityStoreContextImpl<Activity : FragmentActivity> : ActivityS
 
     override fun send(effect: Activity.() -> Unit) {
         // We allow emitting effects only after activity has started
-        startedActivity()?.effect() ?: run {
-            // Log missing activity.
+        if (Utils.isMainThread()) {
+           startedActivity()?.effect()
+        } else {
+            Utils.mainThreadHandler.post {
+                startedActivity()?.effect()
+            }
```

Comment:
Is there a chance that all effects happen before the activity has started and therefore all effects are lost and never emitted?

Reply:
Should they be stored somewhere until the activity is started and then one it is, propagated? I am not sure if this ever even would happen.

Reply:
We would miss any effect that happens before the activity is started. This has been the behavior since this API was introduced, I haven't seen any problems with this as most issues would happen if somebody emits an effect during activity initialization.